pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +333,83 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{
+		LabelSelector: label.TopologyNetwork.Name,
```

Comment:
This will open a new watch, when we already have one. Should prefer to just filter on the clientside IMO